src/Utilities/cachingoptimizer.jl

```diff
@@ -113,14 +113,24 @@ mode(m::CachingOptimizer) = m.mode
 """
     reset_optimizer(m::CachingOptimizer, optimizer::MOI.AbstractOptimizer)
 
-Sets or resets `m` to have the given empty optimizer. Can be called
-from any state. The `CachingOptimizer` will be in state `EMPTY_OPTIMIZER` after the call.
+Sets or resets `m` to have the given empty optimizer `optimizer`.
+
+Can be called from any state. An assertion error will be thrown if `optimizer`
+is not empty.
+
+The `CachingOptimizer` `m` will be in state `EMPTY_OPTIMIZER` after the call.
 """
 function reset_optimizer(m::CachingOptimizer, optimizer::MOI.AbstractOptimizer)
     @assert MOI.is_empty(optimizer)
     m.optimizer = optimizer
     m.state = EMPTY_OPTIMIZER
     for attr in MOI.get(m.model_cache, MOI.ListOfOptimizerAttributesSet())
+        # Skip attributes which don't apply to the new optimizer.
+        if attr isa MOI.RawParameter
```

Comment:
So if I undersand correctly, this is a workaround for solvers not implementing `supports` correctly for RawParameter? Otherwise, this would we caughr by the second condition

Reply:
Or is it because RawParameter with the same name might have different meaning?

Reply:
This should be clarified in a comment

Reply:
> because RawParameter with the same name might have different meaning?

This.